src/features/island/bumpkin/components/BumpkinSkillsModal.tsx

```diff
@@ -0,0 +1,242 @@
+import seedSpecialist from "assets/skills/seed_specialist.png";
+import leftArrow from "assets/icons/arrow_left.png";
+import { Label } from "components/ui/Label";
+import { Button } from "components/ui/Button";
+import { getAvailableBumpkinSkillPoints } from "features/game/events/landExpansion/pickSkill";
+import { Context } from "features/game/GameProvider";
+import { useActor } from "@xstate/react";
+import { getKeys } from "features/game/types/craftables";
+import classNames from "classnames";
+import { SkillPath } from "./SkillPath";
+import { SkillPointsLabel } from "./SkillPointsLabel";
+import { acknowledgeSkillPoints } from "../lib/skillPointStorage";
+
+export const BumpkinSkillsModal: React.FC<{ onClose: () => void }> = ({
+  onClose,
+}) => {
+  const { gameService } = useContext(Context);
+  const [
+    {
+      context: { state },
+    },
+  ] = useActor(gameService);
+
+  const [selectedSkill, setSelectedSkill] =
+    useState<BumpkinSkillName>("Green Thumb");
+  const [selectedSkillPath, setSelectedSkillPath] =
+    useState<BumpkinSkillTree | null>(null);
+  const [skillsInPath, setSkillsInTree] = useState<BumpkinSkill[]>([]);
+  const [showConfirmButton, setShowConfirmButton] = useState(false);
+
+  const onSkillCategoryClickHandler = (category: BumpkinSkillTree) => {
+    setSelectedSkillPath(category);
+
+    const skillsInCategory: BumpkinSkill[] = getSkills(category);
+
+    setSkillsInTree(skillsInCategory);
+
+    const nextAvailableSkillInTree =
+      skillsInCategory.find((skill) => {
+        return !(`${skill.name}` in { ...state.bumpkin?.skills });
+      }) ?? skillsInCategory[0];
+
+    const defaultSkill = nextAvailableSkillInTree ?? skillsInCategory[0];
+
+    setSelectedSkill(defaultSkill.name);
+  };
+
+  const { bumpkin } = state;
+
+  const availableSkillPoints = getAvailableBumpkinSkillPoints(bumpkin);
+  const hasSelectedSkill = !!bumpkin?.skills[selectedSkill];
+
+  const { points: pointsRequired, skill: skillRequired } =
+    BUMPKIN_SKILL_TREE[selectedSkill].requirements;
+
+  const requiredSkillImage = skillRequired
+    ? BUMPKIN_SKILL_TREE?.[skillRequired].image
+    : undefined;
+
+  const missingSkillRequirement = skillRequired
+    ? !getKeys({ ...bumpkin?.skills }).includes(skillRequired)
+    : false;
+
+  const missingPointRequirement = availableSkillPoints < pointsRequired;
+
+  const handleClaim = () => {
+    setShowConfirmButton(false);
+    gameService.send("skill.picked", { skill: selectedSkill });
+    acknowledgeSkillPoints(gameService.state.context.state.bumpkin);
+  };
+
+  return (
+    <>
+      <Panel className="pt-5 relative">
+        <div className="flex justify-between absolute top-1.5 left-0.5 right-0 items-center">
+          <div className="flex">
+            <Tab isActive>
+              <img src={seedSpecialist} className="h-5 mr-2" />
+              <span className="text-sm text-shadow">Skills</span>
+            </Tab>
+          </div>
+          <img
```

Comment:
<img width="176" alt="Screen Shot 2022-09-29 at 2 06 31 pm" src="https://user-images.githubusercontent.com/11745561/192937062-96922738-9a2c-42cf-b857-01b9acffb54b.png">

This button doesn't seem to do anything for me 🤔

Reply:
Hmmmmm I can't recreate this. It's closing for me. I will look into it.